zk_toolbox/crates/zk_supervisor/src/commands/test/all.rs

```diff
@@ -0,0 +1,95 @@
+use std::thread;
+
+use common::{cmd::Cmd, config::global_config, logger};
+use xshell::{cmd, Shell};
+
+use super::{
+    args::{
+        all::AllArgs, integration::IntegrationArgs, recovery::RecoveryArgs, revert::RevertArgs,
+        upgrade::UpgradeArgs,
+    },
+    integration, recovery, revert, upgrade,
+};
+
+pub fn run(shell: &Shell, args: AllArgs) -> anyhow::Result<()> {
+    logger::info("Run integration tests");
+    integration::run(
+        shell,
+        IntegrationArgs {
+            external_node: false,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run recovery tests (from snapshot)");
+    recovery::run(
+        shell,
+        RecoveryArgs {
+            snapshot: true,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run recovery tests (from genesis)");
+    recovery::run(
+        shell,
+        RecoveryArgs {
+            snapshot: false,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run external-node");
+    let _handle = thread::spawn(move || {
+        let chain = global_config().chain_name.clone();
+
+        let server_shell = Shell::new().unwrap();
```

Comment:
clone of the original shell will behave the same

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2594/commits/c61609c0c279385d23dbff2b690a6b32ff41a02c